pkg/ebpf/tracee.go

```diff
@@ -1346,6 +1355,126 @@ func (t *Tracee) initBPF() error {
 	return errfmt.WrapError(err)
 }
 
+func clearMap(bpfMap *bpf.BPFMap) error {
+	var err error
+
+	var iter = bpfMap.Iterator()
+	for iter.Next() {
+		addr := binary.LittleEndian.Uint64(iter.Key())
+		err = bpfMap.DeleteKey(unsafe.Pointer(&addr))
+
+		if err != nil {
+			logger.Error("err occurred DeleteKey: " + err.Error())
+			return err
+		}
+	}
+
+	return nil
+}
+
+func clearState(unhiddenModsMap *bpf.BPFMap, modsFromProcfs *bpf.BPFMap) {
+	_ = clearMap(unhiddenModsMap)
+	_ = clearMap(modsFromProcfs)
+}
+
+func (t *Tracee) lkmSeekerRoutine() {
+	if t.events[events.HiddenKernelModule].emit == 0 {
+		return
+	}
+
+	err := derive.InitFoundHiddenModulesCache()
+	if err != nil {
+		logger.Error("err occurred initializing kernel hidden modules cache: " + err.Error())
+		return
+	}
+
+	unhiddenMap, err := t.bpfModule.GetMap("unhidden_modules_map")
+	if err != nil {
+		logger.Error("err occurred GetMap: " + err.Error())
+		return
+	}
+
+	modulesFromProcFs, err := t.bpfModule.GetMap("modules_from_procfs")
+	if err != nil {
+		logger.Error("err occurred GetMap: " + err.Error())
+		return
+	}
+
+	for {
+		clearState(unhiddenMap, modulesFromProcFs)
+		err = t.fillModulesFromProcFs(modulesFromProcFs)
+		if err != nil {
+			logger.Error("hidden kernel module seeker stopped!")
+			return
+		}
+
+		t.triggerKernelModuleSeeker()
+
+		r := rand.Intn(300) + 10 // min 10 seconds sleep
+		time.Sleep(time.Duration(r) * time.Second)
+	}
+}
+
+func (t *Tracee) fillModulesFromProcFs(modulesFromProcFs *bpf.BPFMap) error {
```

Comment:
It seems that some operations here require bumping the effective capabilites.
For that, you should use the capabilities Requested() func and test when capabilitiies bypass=false (soon this will become the default)